src/frontend/sharedComponents/icons/CategoryIcon.tsx

```diff
@@ -45,13 +47,25 @@ export const CategoryIcon = React.memo<IconProps>(
 );
 
 export const CategoryCircleIcon = ({
-  style,
+  color,
+  iconId,
   size = "medium",
-  ...props
-}: IconProps) => {
+}: CategoryCircleIconProps) => {
+  const [{ presets }] = React.useContext(ConfigContext);
+
+  // If the preset defines a "color" field for *any* point-based category
+  // we want to render a thicker border, even if not all categories have a color defined
+  const presetsUseColors = Array.from(presets.values()).some(
+    p => p.geometry.includes("point") && !!p.color
+  );
+
   return (
-    <Circle radius={radii[size]} style={style}>
-      <CategoryIcon {...props} size={size} />
+    <Circle
+      color={color}
+      radius={radii[size]}
+      style={presetsUseColors ? { borderWidth: 1.5 } : undefined}
+    >
+      <CategoryIcon iconId={iconId} size={size} />
```

Comment:
Since the variable border width is only applicable to the circle icon when used for categories, figured it would make more sense to do that calculation here and provide as a style override to the default circle border width if needed, removing any of that logic in the Circle component itself

Reply:
@gmaclennan is the `presetsUseColors` calculation the proper way of determining if the preset has a color defined that's applicable to categories?